### DISCUSSION TITLE
How can I change page title when I use full page component

### DISCUSSION THREAD

**Discussion by ShekhSaifuddin007** [livewire/livewire#1755] [4. General]
<head>
          <title>title- {{ config('app.name') }}</title>
    </head>

**Comment by ppegu** [comment]
are you talking about to make the page title dynamic for every page ? then you can do something like `@section('title', config('app.name').'-' .ucfirst(Route::currentRouteName()))
`

**Comment by orware** [comment]
I have a similar question/issue but it's more involved.

As part of my layout I have the following (which was a suggestion I found elsewhere online somewhere):
```
<title>@hasSection('page-title') @yield('page-title') - {{ config('app.name') }} @else {{ config('app.name') }} @endif</title>
```

And then in my Livewire components I have the following:
```
@section('page-title')
    @if(empty($title))
        Home
    @else
        {{ $title }}
    @endif
@endsection
```

The main issue I have is that this works very well for a full page load, but if I go from let's say the "Home" to a subpage via a Livewire action, it doesn't actually update the title correctly and ends up staying as-is. This is also further complicated by the fact that normally the title for my Livewire components can be dynamic since if a user runs a search on the page, that would update the title as well (with a more regular approach that resulted in full page loads and the title updating correctly, but with Livewire the title would stay as it was when the page initially loaded).

There are some older issues that seem to talk about the topic:
https://github.com/livewire/livewire/issues/1835 (the suggestion in here to pass in the title to the layout didn't really seem to change any of the behavior I was observing so that one didn't seem to solve this particular issue)
https://github.com/livewire/livewire/issues/1756

But from some of the responses it doesn't seem like there is full support available for dynamically adjusting the `<title>` element out of the box in Livewire in a really easy way (although that would be a nice addition).

Some potential drawbacks for not having this behave as one might expect out of the box is that for tools that may be collecting usage analytics on your site, they may not correctly collect data for those searches and situations where the title should technically be changing (so instead, those analytics would likely all group those visits under something like the "Home" title instead).

@calebporzio, since there is already support for maintaining the browser history automatically when using Livewire, I'm hoping there might be a way to integrate having the title also be dynamic in some easier way so that we can maintain that extra detail in the browser without having to jump through extra hoops as developers to make that capability work as we might want it to (either that or have a pattern for this option potentially provided in the docs so we can adopt that solution).
